packages/app-headless-cms/src/admin/components/ContentEntries/Filters/RefFieldRenderer/components/RefPresenter.ts

```diff
@@ -0,0 +1,76 @@
+import { makeAutoObservable, runInAction } from "mobx";
+import { EntryDTO, EntryRepository } from "../domain";
+
+interface EntryOption {
+    id: string;
+    name: string;
+}
+
+export interface RefPresenterViewModel {
+    selected?: EntryOption;
+    options: EntryOption[];
+    loading: boolean;
+}
+
+interface IRefPresenter {
+    load(id?: string): Promise<void>;
+    search(query: string): Promise<void>;
+    get vm(): RefPresenterViewModel;
+}
+
+export class RefPresenter implements IRefPresenter {
+    private repository: EntryRepository;
+    private currentEntry: EntryDTO | null = null;
+
+    constructor(repository: EntryRepository) {
+        this.repository = repository;
+        makeAutoObservable(this);
+    }
+
+    async load(value?: string) {
+        try {
+            if (!value) {
+                return;
+            }
+
+            const { entryId, modelId } = JSON.parse(value);
```

Comment:
Add a comment about _why_ is `value` a string, and not an object? Because it's not clear why you need to JSON.parse if you doin't know all the internals.